freva-rest/src/databrowser_api/endpoints.py

```diff
@@ -217,20 +138,20 @@ async def extended_search(
 async def data_search(
```

Comment:
That's the get method definition for the rest server. I don't even know if this would work if this method would
be a contextmanager. Also, what would you like to achieve with such a construct?

Reply:
actually using `asynccontextmanager` for the GET method definition in freva rest might seem unconventional but if we want to be sure the database connection is opened and closed properly, regardless of whether the database query was successful or raised an exception or defining another func for closing the db after usage, it would be a neat idea. for example without using this decorator, we have to manually ensure that database connections are closed after their use. but no worries it was just a small idea to make the code more standard, can be ignored :). review got resolved

Reply:
Hmm, I don't know how the context manager would get called within the fast API framework. I mean fast API would have to involve a `with` statement or something similar in order to make this work.

But regardless or the context manager, I think it's not a good idea to create a new MongoDB connection every time a request is made. Reusing existing connections is way more efficient. And the store_result method actually takes care about non existing database connections and so on.

Reply:
true, we need `with` statement to context manage the flow and is not straight forward in FastAPI. 
And again you r right, creating new connection for each req is not efficient, pooling and reusing an existed connection using [motor](https://github.com/FREVA-CLINT/databrowserAPI/blob/8d142b19341a94be49399cddb2d9e1a5376b016b/src/databrowser/config.py#L34C7-L34C19) was such a smart idea. 

And I'm looking at `store_results` func of core.py, but i was wondering, where this func can take care of none existing db conns? what would be ur idea to add a try except to ping the conn like `await self._config.mongo_collection.database.command('ping')` before `insert_one` to be sure the conn is alive or existed to pool and insert. 
https://www.mongodb.com/docs/drivers/motor/#connect-to-mongodb-atlas